From b734c1223b1de1761eccf08e8a216af54c375826 Mon Sep 17 00:00:00 2001
From: Zyta Szpak <zr@semihalf.com>
Date: Tue, 27 Feb 2018 14:01:50 +0100
Subject: [PATCH 22/74] mrvlna: add header

---
 drivers/net/mrvlna/mrvlna_ethdev.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/net/mrvlna/mrvlna_ethdev.h b/drivers/net/mrvlna/mrvlna_ethdev.h
index 68903c7..2cf925a 100644
--- a/drivers/net/mrvlna/mrvlna_ethdev.h
+++ b/drivers/net/mrvlna/mrvlna_ethdev.h
@@ -47,6 +47,7 @@
 
 #include <drivers/mv_neta.h>
 #include <drivers/mv_neta_bpool.h>
+#include <drivers/mv_neta_ppio.h>
 
 //TODO check all the defines and offsets!!
 /** Packet offset inside RX buffer. */
-- 
2.7.4

