From 710807b95ca1b3eb4848e713271c23e982211e9e Mon Sep 17 00:00:00 2001
From: Zyta Szpak <zr@semihalf.com>
Date: Wed, 28 Feb 2018 10:29:37 +0100
Subject: [PATCH 29/74] mrvlna: correct link state update callback

---
 drivers/net/mrvlna/mrvlna_ethdev.c | 12 ++++--------
 1 file changed, 4 insertions(+), 8 deletions(-)

diff --git a/drivers/net/mrvlna/mrvlna_ethdev.c b/drivers/net/mrvlna/mrvlna_ethdev.c
index f9a48cc..fe9d4d1 100644
--- a/drivers/net/mrvlna/mrvlna_ethdev.c
+++ b/drivers/net/mrvlna/mrvlna_ethdev.c
@@ -692,8 +692,8 @@ mrvlna_link_update(struct rte_eth_dev *dev, int wait_to_complete __rte_unused)
 	case SPEED_1000:
 		dev->data->dev_link.link_speed = ETH_SPEED_NUM_1G;
 		break;
-	case SPEED_10000:
-		dev->data->dev_link.link_speed = ETH_SPEED_NUM_10G;
+	case SPEED_2500:
+		dev->data->dev_link.link_speed = ETH_SPEED_NUM_2_5G;
 		break;
 	default:
 		dev->data->dev_link.link_speed = ETH_SPEED_NUM_NONE;
@@ -703,12 +703,8 @@ mrvlna_link_update(struct rte_eth_dev *dev, int wait_to_complete __rte_unused)
 							 ETH_LINK_HALF_DUPLEX;
 	dev->data->dev_link.link_autoneg = edata.autoneg ? ETH_LINK_AUTONEG :
 							   ETH_LINK_FIXED;
-	/*
-	 * TODO this function is not implemented in MUSDK
-	 * find other way to check link status
-	 */
-	/*neta_ppio_get_link_state(priv->ppio, &link_up);*/
-	link_up = 1;
+
+	neta_ppio_get_link_state(priv->ppio, &link_up);
 	dev->data->dev_link.link_status = link_up ? ETH_LINK_UP : ETH_LINK_DOWN;
 
 	return 0;
-- 
2.7.4

