From e97f9bfeb1aac53a76a64bde6bca2aa1132a6224 Mon Sep 17 00:00:00 2001
From: Zyta Szpak <zr@semihalf.com>
Date: Tue, 13 Mar 2018 21:16:35 +0100
Subject: [PATCH 48/74] mrvlna: fix buffer address

---
 drivers/net/mrvlna/mrvlna_ethdev.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/net/mrvlna/mrvlna_ethdev.c b/drivers/net/mrvlna/mrvlna_ethdev.c
index 264dd79..cbd80f0 100644
--- a/drivers/net/mrvlna/mrvlna_ethdev.c
+++ b/drivers/net/mrvlna/mrvlna_ethdev.c
@@ -207,7 +207,8 @@ mrvlna_recv_buffs_free(struct neta_rxq *rxq, struct neta_ppio_desc *desc)
 
 	for (i = 0; i < size; i++) {
 		if (desc) {
-			addr = neta_ppio_inq_desc_get_cookie(desc);
+			addr = cookie_addr_high |
+					neta_ppio_inq_desc_get_cookie(desc);
 			if (addr)
 				rte_pktmbuf_free((struct rte_mbuf *)addr);
 			desc++;
@@ -292,9 +293,10 @@ mrvlna_rx_queue_flush(struct neta_rxq *rxq, struct neta_ppio_desc *descs)
 		ret = neta_ppio_recv(rxq->priv->ppio,
 					rxq->queue_id,
 					descs, (uint16_t *)&num);
+		if (num)
+			mrvlna_recv_buffs_free(rxq, descs);
 	} while (ret == 0 && num);
 
-	mrvlna_recv_buffs_free(rxq, descs);
 }
 
 /**
-- 
2.7.4

