From 40640961bfc0eea12a6a2336b6d65e58a7650a9c Mon Sep 17 00:00:00 2001
From: Zyta Szpak <zr@semihalf.com>
Date: Tue, 13 Mar 2018 23:01:45 +0100
Subject: [PATCH 49/74] mrvlna: fix ppio_id for subsequent ports

---
 drivers/net/mrvlna/mrvlna_ethdev.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/net/mrvlna/mrvlna_ethdev.c b/drivers/net/mrvlna/mrvlna_ethdev.c
index cbd80f0..83d9dbc 100644
--- a/drivers/net/mrvlna/mrvlna_ethdev.c
+++ b/drivers/net/mrvlna/mrvlna_ethdev.c
@@ -655,6 +655,7 @@ mrvlna_dev_configure(struct rte_eth_dev *dev)
 	/* Default: 1 TC, no QoS supported. */
 	priv->ppio_params.inqs_params.num_tcs = 1;
 	priv->ppio_params.inqs_params.tcs_params[0].pkt_offset = MRVL_NETA_PKT_OFFS;
+	priv->ppio_id = dev->data->port_id;
 	snprintf(match, sizeof(match), "eth%d", priv->ppio_id);
 	priv->ppio_params.match = match;
 	/* TODO check if DPDK has already set mtu to default value */
-- 
2.7.4

