From 335e65034d44ebfc2a6178e403330212f7f27eca Mon Sep 17 00:00:00 2001
From: Zyta Szpak <zr@semihalf.com>
Date: Sun, 18 Mar 2018 16:26:59 +0100
Subject: [PATCH 62/74] net/mrvlna: free mbuf in case of erroneous rx pkt

---
 drivers/net/mrvlna/mrvlna_ethdev.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/net/mrvlna/mrvlna_ethdev.c b/drivers/net/mrvlna/mrvlna_ethdev.c
index 2ed10ca..10135d0 100644
--- a/drivers/net/mrvlna/mrvlna_ethdev.c
+++ b/drivers/net/mrvlna/mrvlna_ethdev.c
@@ -551,7 +551,9 @@ mrvlna_rx_pkt_burst(void *rxq, struct rte_mbuf **rx_pkts, uint16_t nb_pkts)
 		/* drop packet in case of mac, overrun or resource error */
 		status = neta_ppio_inq_desc_get_l2_pkt_error(&descs[i]);
 		if (unlikely(status != NETA_DESC_ERR_OK)) {
-			/* TODO free the mbuf since it was taken but it won't be used */
+			/* Release the mbuf to the mempool since
+			 * it won't be transferred to tx path */
+			rte_pktmbuf_free(mbuf);
 			q->drop_mac++;
 			continue;
 		}
-- 
2.7.4

