From 43feb9f7547b3e986c26d15f5b7c31f4af20b0e7 Mon Sep 17 00:00:00 2001
From: Zyta Szpak <zr@semihalf.com>
Date: Tue, 20 Mar 2018 13:15:51 +0100
Subject: [PATCH 67/74] net/mvneta: remove match for unused pool

---
 drivers/net/mvneta/mvneta_ethdev.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/drivers/net/mvneta/mvneta_ethdev.c b/drivers/net/mvneta/mvneta_ethdev.c
index 0a2684f..a25e155 100644
--- a/drivers/net/mvneta/mvneta_ethdev.c
+++ b/drivers/net/mvneta/mvneta_ethdev.c
@@ -1129,15 +1129,12 @@ static const struct eth_dev_ops mvneta_ops = {
 static struct neta_priv *
 mvneta_priv_create(const char *dev_name)
 {
-	char match[MVNETA_MATCH_LEN];
 	struct neta_priv *priv;
 
 	priv = rte_zmalloc_socket(dev_name, sizeof(*priv), 0, rte_socket_id());
 	if (!priv)
 		return NULL;
 
-	snprintf(match, sizeof(match), "pool-%d", priv->pp_id);
-
 	return priv;
 }
 
-- 
2.7.4

